packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -364,12 +365,23 @@ export async function buildExtraViteConfig(
 
 	// handle prebundling for svelte files
 	if (options.prebundleSvelteLibraries) {
-		extraViteConfig.optimizeDeps.extensions = options.extensions ?? ['.svelte'];
-		// Add esbuild plugin to prebundle Svelte files.
-		// Currently a placeholder as more information is needed after Vite config is resolved,
-		// the real Svelte plugin is added in `patchResolvedViteConfig()`
-		extraViteConfig.optimizeDeps.esbuildOptions = {
-			plugins: [{ name: facadeEsbuildSveltePluginName, setup: () => {} }]
+		extraViteConfig.optimizeDeps = {
+			...extraViteConfig.optimizeDeps,
+			// only prebundle for dev, build isn't working with vite-plugin-svelte
+			disabled: 'build',
```

Comment:
this possibly overrides a vite config setting by the user, but i feel like prebundleSvelteLibraries takes precedence here. As we only enable it for dev (which is the current default for vite too) i don't think this is a breaking change, but in theory it could reenable optimizeDeps for someone who had it disabled before.

Add a log for that case just to be sure?

Reply:
I don't think we should be blocking users if they want to prebundle in build (even if it doesn't work). They could be interested in debugging this feature so we won't prevent them from doing so.

Providing logs would be great though if we detect that they enable prebundle in build. We could say that we don't support it & dragons etc

Reply:
updated code to only enable prebundleSvelteLibraries in development by default and logging warnings for problematic combinations of vite optimizeDeps config values